### PR TITLE
core: allow to change deposit's `till` for owner only

### PR DESCRIPTION
**Pull request by AnnaShaleva**
### Problem

Anyone can change deposit's `till` for free.

### Solution

Set the default value in `onPayment` method.

### CODE REVIEW

File: pkg/core/native/notary.go [hunk 1]
```diff
@@ -158,23 +159,31 @@ func (n *Notary) onPayment(ic *interop.Context, args []stackitem.Item) stackitem
...
 			Amount: new(big.Int),
 		}
+		if !allowedChangeTill {
+			till = currentHeight + defaultDepositDeltaTill
+		}
 	} else {
-		if till < deposit.Till {
-			panic(fmt.Errorf("`till` shouldn't be less then the previous value %d", deposit.Till))
+		if allowedChangeTill { // only deposit's owner is allowed to set or update `till`
+			if till < currentHeight {
```
Comment (roman-khimov) [inline pkg/core/native/notary.go:179]: It should either be a generic (relative to `deposit == nil`) check or not exist at all, it doesn't affect correctness of this contract and it's hard to tell what is better for the user here, failed or partially successful (deposit accepted, but not really usable) transaction.

Comment (AnnaShaleva) [reply pkg/core/native/notary.go]: So then maybe we'll set the default `till` value in `onPayment` method, and then the owner will be able to update `till` value via `lockDepositUntil`?

Comment (roman-khimov) [reply pkg/core/native/notary.go]: The majority of invocations will have `to == from` and in this case it's quite natural to allow setting `till`.

Comment (AnnaShaleva) [reply pkg/core/native/notary.go]: OK, I rewrote this part, now `till` check is generic.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**